collector/cmd/dcos-metrics-collector/config.go

```diff
@@ -8,18 +8,35 @@ import (
 	yaml "gopkg.in/yaml.v2"
 )
 
-type ConfigFile struct {
-	PollAgent     bool `yaml:"poll_agent"`
-	HttpProfiler  bool `yaml:"http_profiler"`
-	KafkaProducer bool `yaml:"kafka_producer"`
-	ConfigPath    string
+type MasterConfig struct {
```

Comment:
The linter is going to complain about a lot of things at the moment. I have a separate commit that fixes dozens of these cases, but we're waiting until after this PR is reviewed and merged until I apply it 😄